File: fixtures/scale/dart/large/lib/ciphers/djb2/djb2_hash.dart
import '../../core/hash_function.dart';
import '../../util/bytes.dart';

/// Bernstein's hash: state * 33 + byte.
class Djb2Hash implements HashFunction {
  @override
  String name() => 'djb2';

  @override
  int digest(String input) {
    var state = 5381;
    for (final b in bytesOf(input)) {
      state = (((state << 5) + state) + b) & 0xFFFFFFFF;
    }
    return state;
  }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/djb2/Djb2Hash.kt
package cryptobench.ciphers.djb2

import cryptobench.core.HashFunction
import cryptobench.util.Bytes

/** Bernstein's hash: state * 33 + byte. */
class Djb2Hash : HashFunction {

    override fun name(): String = "djb2"

    override fun digest(input: String): Long {
        var state = 5381L
        for (b in Bytes.of(input)) {
            state = ((state shl 5) + state) + (b.toLong() and 0xFF)
        }
        return state
    }
}
